### PR TITLE
feat(law-practice): publish court reporter vocabulary

### PR DESCRIPTION
**Pull request by kriegcloud**
## Local proof

- publish:head-install-preflight: passed
- publish:git:push: passed

Verdict: .beep/yeet/runs/feat_court-reporter-vocabulary-62d8e9d0e784/verdict.json

---

## Provenance

- Branch: <code>feat/court-reporter-vocabulary</code>
- Harness: `codex`

---
<a href="https://app.blacksmith.sh/beep-effect/codesmith/beep-effect/pr/865"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1790433703&installation_model_id=424656&pr_number=865&repository=beep-effect%2Fbeep-effect&return_to=https%3A%2F%2Fgithub.com%2Fbeep-effect%2Fbeep-effect%2Fpull%2F865&signature=b2fde1f51dabe45b4c4e18ab51231c20a71fcb90e56fa7f43b08886423cdf24e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

File: packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.data.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+ * @category models
+ * @since 0.0.0
+ */
+export const CourtReporterArtifact = CourtReporterArtifactContract.make({
+  schemaVersion: CourtVocabulary.schemaVersion,
+  projectionVersion: CourtVocabulary.projectionVersion,
+  artifactVersion: CourtVocabulary.artifactVersion,
+  policy: CourtReporterCompatibilityPolicy,
+  courts: CourtVocabulary,
+  reporters: ReporterVocabulary,
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.data.ts:74]: Addressed in 2e8b1af8ee: the combined contract now has a schema-level version-coherence check requiring the top-level, court, and reporter artifact versions to match. The focused regression test rejects mismatched independently generated artifacts.

File: packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+    )
+  );
+
+  const successorGroups = pipe(
+    next,
+    A.map((record) =>
+      record.status === "tombstone"
+        ? pipe(
+            record.successorId,
+            O.map((successorId) => [successorId, record.id] as const)
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts:383]: Addressed in 2e8b1af8ee: retained records now emit an incompatible successorRemoval whenever a published successor is removed or replaced, with focused lifecycle coverage.

File: packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+});
+
+const normalizeReporter = (record: ReporterVocabularyRecord): NormalizedRecord => ({
+  id: record.id,
+  semanticKey: record.semanticKey,
+  lineageKey: record.lineageKey,
+  aliases: normalizeAliases(record.aliases, record.contextualAliases),
+  rangeCount: A.length(record.editions),
+  status: record.status,
+  successorId: record.successorId,
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts:147]: Addressed in 2e8b1af8ee: normalized effective ranges and reporter editions now retain their concrete boundary values, and changed boundaries are classified as incompatible semantic reuse. Focused tests exercise both court and reporter boundary drift.

File: packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+  normalizeStringSet
+);
+
+const normalizeOption = <A>(value: O.Option<A>): A | null => O.getOrNull(value);
+
+const normalizeCourt = (record: CourtVocabularyRecord): NormalizedRecord => ({
+  id: record.id,
+  semanticKey: record.semanticKey,
+  lineageKey: record.lineageKey,
+  aliases: normalizeAliases(record.aliases, record.contextualAliases),
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts:102]: Addressed completely in 430c4ac6ae. Plain and contextual alias categories are compared independently, including direct category transitions, with regression coverage for contextual-plus-plain and plain-only/contextual-only cases.

File: packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+  const removals = normalizeStringSet([
+    ...A.filter(previous.aliases, (alias) => !A.contains(next.aliases, alias)),
+    ...A.filter(
+      previous.plainAliases,
+      (alias) =>
+        !A.contains(next.plainAliases, alias) &&
+        A.contains(next.aliases, alias) &&
+        A.contains(previousContextualAliases, alias) &&
+        A.contains(nextContextualAliases, alias)
+    ),
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply packages/law-practice/domain/src/values/CourtReporterVocabulary/CourtReporterVocabulary.compatibility.ts]: Fixed in 430c4ac6ae. Direct plain-only to contextual-only and contextual-only to plain-only transitions now emit both the category addition and the incompatible category removal. The new focused regression test covers both directions; 13/13 vocabulary tests and the package check pass.
